### DISCUSSION TITLE
Mixed Content: Request has been blocked, the content must be served over HTTPS.

### DISCUSSION THREAD

**Discussion by cheadevit** [Laravel-Backpack/community-forum#411] [Q&A (Help)]
Hello Guys,

I got some error during I try to deploy my project to the server, 

Here is the version

### PHP VERSION:
PHP 8.1.13 (cli) (built: Nov 26 2022 14:07:36) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.13, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.13, Copyright (c), by Zend Technologies

### LARAVEL VERSION:
v9.52.2

### BACKPACK PACKAGE VERSIONS:
backpack/crud: 5.5.0
backpack/generators: 3.3.14
backpack/pro: 1.6.0

Here is the error message

``` 
Mixed Content: The page at 'https://example.com/login/' was loaded over HTTPS, but requested an insecure script 'http://example.com/js/app.js'. This request has been blocked; the content must be served over HTTPS.
example.com/:21
```

```
Mixed Content: The page at 'https://example.com/login/' was loaded over HTTPS, but requested an insecure stylesheet 'http://example.com/vendors/dist/css/adminlte.min.css'. This request has been blocked; the content must be served over HTTPS.
```

**Comment by o15a3d4l11s2** [comment]
You should activate (and enforce) HTTPS for any content. There are different options available:

1. Using `.htaccess` (I recommend this)
```
RewriteEngine On
# Redirect to HTTPS
RewriteCond %{HTTPS} !=on
RewriteRule ^ https://%{HTTP_HOST}%{REQUEST_URI} [L,R=301]
```
2. Using Laravel - there are a lot of articles online

Also do not forget to set the url with `https` in your `.env` file.

**Reply by cheadevit** [reply]
Thank you, I tried but it's not working, and URL is https already.

**Reply by o15a3d4l11s2** [reply]
I checked your website and the problem now is that `app.js` is not found (not server) at all. This is a different issue.

Here are some follow-up questions regarding your current issue:
1. When you deploy to the server, does the deploy also contain all the assets, or you compile them on the server?
    1. If you deploy with pre-build assets, then run the commands from point 2. and then deploy the compiled assets to your server
    2. If you deploy without pre-build assets, then run the commands from point 2. on your server after a deploy.
2. Do you use Laravel Mix or Vite? Depending on this you will have to run
    1. Laravel Mix: `npm run dev` (for development) or `npm run prod` (for production)
    2. Vite: `npm run dev` and  maybe `rpm run build` (for production)

Also verify if the file `app.js` exists in your Laravel project's folder -> public/js/app.js. Once you compile the assets, it should exist.

**Reply by o15a3d4l11s2** [reply]
Did you get a change to check the points that I mentioned? Feel free to get back to me if you have any other questions/comments.

**Reply by cheadevit** [reply]
I got it,  Thank you for your response

**Reply by cheadevit** [reply]
I got an error after running this command 

Step 21/23 : RUN npm run build
 ---> Running in d45180ddf9b6

> build
> vite build

[91mfile:///app/node_modules/vite/bin/vite.js:7
    await import('source-map-support').then((r) => r.default.install())
    ^^^^^

**Reply by o15a3d4l11s2** [reply]
What node version do you have? Maybe try updating to the latest one first and then retry the commands.

**Reply by cheadevit** [reply]
I think it should node the latest version, and I using docker and here is my Docker file,

```
FROM php:8.1
RUN apt-get update -y && apt-get install -y cron nodejs npm openssl zip unzip git libonig-dev libzip-dev libgd-dev && docker-php-ext-install pdo_mysql gd
RUN docker-php-ext-configure gd --enable-gd --with-freetype --with-jpeg
RUN docker-php-ext-install zip
RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
RUN docker-php-ext-install pdo mbstring zip
RUN cd /usr/local/etc/php/conf.d/ && \
    echo 'memory_limit = 64M' >> /usr/local/etc/php/conf.d/docker-php-memlimit.ini && \
    echo 'post_max_size = 64M' >> /usr/local/etc/php/conf.d/docker-php-memlimit.ini && \
    echo 'upload_max_filesize = 64M' >> /usr/local/etc/php/conf.d/docker-php-memlimit.ini && \
    echo 'max_execution_time = 600' >> /usr/local/etc/php/conf.d/docker-php-memlimit.ini

WORKDIR /app
COPY . /app
RUN mkdir -p /storage/app/public/uploads/
RUN chmod -R 777 /storage/app/public/uploads/
# RUN apt-get install php7.0-g


# RUN composer install
# RUN php artisan cache:clear
RUN composer install --ignore-platform-reqs
RUN php artisan optimize
RUN php artisan storage:link
RUN php artisan route:cache
RUN php artisan route:clear

RUN npm install
RUN npm run dev

RUN php artisan --force key:generate
RUN php artisan --force migrate:fresh
# RUN php artisan --force migrate
RUN php artisan --force db:seed


CMD php artisan serve --host 0.0.0.0 --port=8181
EXPOSE 8181
```

**Reply by o15a3d4l11s2** [reply]
Could you verify the exact node and npm versions that are running inside Docker? Last error seems to me that your node version is lower than 16.

**Comment by jcastroa87** [comment]
This issue will be close due to not activity, please feel free to reopen or create a new one if needed.

Cheers.
